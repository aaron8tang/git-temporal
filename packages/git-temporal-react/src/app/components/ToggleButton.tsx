import React from 'react';
import { style } from 'app/styles';

export interface ToggleButtonProps {
  // This is the text or JSX that gets wrapped in a Toggle Button
  isSelected: boolean;
  onClick: (evt) => void;
  children?: string | JSX.Element | JSX.Element[];
  style?: string | object;
}

export const ToggleButton = (props: ToggleButtonProps): JSX.Element => {
  const styles: any = [];
  if (props.isSelected) {
    styles.push('selected');
  } else {
    styles.push('selectable');
  }
  styles.push(props.style);
  return (
    <div style={style(styles, props.style)} onClick={props.onClick}>
      {props.children || ' '}
    </div>
  );
};
