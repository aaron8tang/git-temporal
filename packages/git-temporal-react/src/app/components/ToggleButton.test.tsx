import React from 'react';
import { shallow } from 'enzyme';

import { ToggleButton } from './ToggleButton';

describe('components/ToggleButton', () => {
  describe('when rendered with isSelected=false', () => {
    let wrapper;
    const onClickMock = jest.fn();
    beforeAll(() => {
      wrapper = shallow(
        <ToggleButton isSelected={false} onClick={onClickMock}>
          ...
        </ToggleButton>
      );
    });
    test('it should match snapshot (should have rendered as selectable)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with isSelected=true', () => {
    let wrapper;
    const onClickMock = jest.fn();
    beforeAll(() => {
      wrapper = shallow(
        <ToggleButton isSelected={true} onClick={onClickMock}>
          ...
        </ToggleButton>
      );
    });
    test('it should match snapshot (should have rendered as selected)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
