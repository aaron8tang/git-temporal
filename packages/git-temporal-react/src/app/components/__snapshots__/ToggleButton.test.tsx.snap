// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ToggleButton when rendered with isSelected=false it should match snapshot (should have rendered as selectable) 1`] = `
<div
  onClick={[MockFunction]}
  style={Object {}}
>
  ...
</div>
`;

exports[`components/ToggleButton when rendered with isSelected=true it should match snapshot (should have rendered as selected) 1`] = `
<div
  onClick={[MockFunction]}
  style={
    Object {
      "background": "lightskyblue",
      "border": "1px solid lightskyblue",
      "borderRadius": 3,
      "cursor": "pointer",
      "userSelect": "none",
    }
  }
>
  ...
</div>
`;
